### DISCUSSION TITLE
[Feature Request] 2FA

### DISCUSSION THREAD

**Discussion by zero77** [nginx-proxy/nginx-proxy#1626] [Ideas]
Can you please add 2FA to provide additional security before clients are able to access web server content.

Google already provides a 2FA module that can be integrated with protocols like SSH and OpenVPN but, there are other 2fa options out there.
https://github.com/google/google-authenticator-libpam

**Comment by tkw1536** [comment]
I am not sure how 2FA would be integrated into nginx (and the linked page doesn't mention anything).

**Comment by tkw1536** [comment]
This project does support [Basic Authentication](https://github.com/nginx-proxy/nginx-proxy#basic-authentication-support), but it is not clear to me how one would extend that into 2FA (TOTP or otherwise). If there is an existing nginx module I guess one could [add a custom config](https://github.com/nginx-proxy/nginx-proxy#custom-nginx-configuration).

**Comment by DatAres37** [comment]
You could use a browser certificate as a second factor for example. Every normal nginx guide should work with the [vhost configuration](https://github.com/nginx-proxy/nginx-proxy#per-virtual_host).

**Comment by RomeSilvanus** [comment]
I just wanted to chime in that this would maybe help:
https://github.com/Arno0x/TwoFactorAuth
This does seem to have nginx integration.
2FA support would be nice for more secure access to applications.

**Comment by buchdag** [comment]
In my opinion 2FA should be provided at the app level, not by nginx-proxy. The way I see it, this project scope is to be a reverse proxy / ingress controller, not to replace proper auth in app that don't implement it, or to supplement the proxied app's feature set.
